### PR TITLE
IAM v2 misc cleanup

### PR DESCRIPTION
**Pull request by msorens**
### :nut_and_bolt: Description: What code changed, and why?

Cleaning up a few vestigial mentions of V2.

### :chains: Related Resources
NA

### :+1: Definition of Done
Cleaner code base.

### :athletic_shoe: How to Build and Test the Change
If buildkite is happy, we should be good.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .studio/automate-gateway [hunk 1]
```diff
@@ -606,8 +606,8 @@ function gateway_grpcurl_get() {
 document "get_api_token" <<DOC
   This will idempotently generate an API token that has enough access for data ingestion.
 
-  Note: if using IAM v2, this token will NOT be an admin token; but one that is a member
-  of the default ingest-access policy.
+  Note: this token will be a member of the default ingest-access policy, NOT an admin token.
```
Comment (susanev) [inline .studio/automate-gateway:609]:
```suggestion
  Note: this token will be a member of the chef-managed ingest-access policy, NOT the chef-managed admin policy.
```

File: components/automate-chef-io/data/docs/cli_chef_automate/commands/chef-automate_iam_token_create.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ options:
 - name: admin
   default_value: "false"
   usage: |
-    Generate a token and grant it admin-level permission (IAM v2)
+    Generate a token and grant it admin-level permission
```
Comment (susanev) [inline components/automate-chef-io/data/docs/cli_chef_automate/commands/chef-automate_iam_token_create.yaml:8]:
```suggestion
    Generate a token and add it to the chef-managed admin policy
```

File: components/automate-cli/cmd/chef-automate/dev.go [hunk 3]
```diff
@@ -947,7 +947,7 @@ func newCreateIAMDevUsersCmd() *cobra.Command {
 	return &cobra.Command{
 		Use:   "create-iam-dev-users",
 		RunE:  runCreateIAMDevUsersCmd,
-		Short: `Create IAM v2 dev users ("viewer" and "editor") idempotently`,
+		Short: `Create dev users ("viewer" and "editor") idempotently`,
```
Comment (susanev) [inline components/automate-cli/cmd/chef-automate/dev.go:950]: i think IAM is still useful to say here..no?

File: components/automate-cli/cmd/chef-automate/iam.go [hunk 4]
```diff
@@ -59,7 +59,7 @@ func newIAMCreateTokenCommand() *cobra.Command {
 		&iamCmdFlags.adminToken,
 		"admin",
 		false,
-		"Generate a token and grant it admin-level permission (IAM v2)")
+		"Generate a token and grant it admin-level permission")
```
Comment (susanev) [inline components/automate-cli/cmd/chef-automate/iam.go:62]:
```suggestion
		"Generate a token and add it to the chef-managed admin policy")
```

File: components/automate-gateway/gateway/middleware/authz/authz.go [hunk 5]
```diff
@@ -50,7 +50,6 @@ func (c *client) Handle(ctx context.Context, subjects []string, projectsToFilter
 		"auth.subjects": subjects,
 		"auth.resource": resource,
 		"auth.action":   action,
-		"iam.version":   "iam_v2",
```
Comment (msorens) [inline components/automate-gateway/gateway/middleware/authz/authz.go:53]: As far as I can tell, this is unused.

**Pull request merged**